noir_stdlib/src/append.nr

```diff
@@ -0,0 +1,33 @@
+// Appends two values together, returning the result.
+//
+// An alternate name for this trait is `Monoid` if that is familiar.
+// If not, it can be ignored.
+//
+// It is expected that for any implementation:
+// - `T::empty().append(x) == x`
+// - `x.append(T::empty()) == x`
+trait Append {
+    fn empty() -> Self;
+    fn append(self, other: Self) -> Self;
+}
+
+impl<T> Append for [T] {
+    fn empty() -> Self {
+        &[]
+    }
+
+    fn append(self, other: Self) -> Self {
+        // Slices have an existing append function which this will resolve to.
+        self.append(other)
+    }
+}
+
+impl Append for Quoted {
```

Comment:
Can you add a test for appending two `Quoted`'s?

Reply:
Looks like there's a bug currently preventing me from doing so: https://github.com/noir-lang/noir/issues/5428